process/sync/shardblock.go

```diff
@@ -290,11 +292,13 @@ func (boot *ShardBootstrap) getMinNotarizedMetaBlockNonceInHeader(
 			return minNonce, err
 		}
 
-		if ni.blockWithLastNotarized[shardId] == 0 && metaBlock.Nonce == ni.lastNotarized[shardId] {
+		if ni.blockWithLastNotarized[shardId] == 0 && ni.lastNotarized[shardId] != nil && metaBlock.Nonce == ni.lastNotarized[shardId].Nonce {
```

Comment:
Extract these conditions in a separate variable. Also for line 300.

Reply:
done